File: scripts/ruby/gps_verification.rb
#!/usr/bin/env ruby
require "pi_piper"

pin = PiPiper::Pin.new(:pin => 27, :direction => :out)

loop do
  result = `ntpq -p`
  lines = result.split("\n")
  reach = lines[2].split("\s")[6].to_i
  if reach>=17 then
    pin.on
  else
    pin.off
  end
  sleep(10)
end

File: scripts/python/gps_verification.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import subprocess
import time

GPIO.setwarnings(False)
GPIO.setmode(GPIO.BCM)
GPIO.setup(27, GPIO.OUT)

while True:
    result = subprocess.run("chronyc sources -v", shell=True, stdout=subprocess.PIPE , stderr=subprocess.PIPE ,encoding="utf-8")
    lines = result.stdout.split("\n")
    for line in lines:
        parse = line.split()
        if len(parse)>1:
            if parse[1] == "GPS":
                if parse[0] == "#*":
                    print("GPS synchronized")
                    GPIO.output(27,1)
                else:
                    print("GPS not synchronized")
                    GPIO.output(27,0)
    time.sleep(30)
